### PR TITLE
UploadSet now has a depositor property.

### PR DESCRIPTION
**Pull request by hectorcorrea**
Addresses #468

### CODE REVIEW

**Comment by hectorcorrea** [comment]
Should I add the new `depositor` field to the UploadSet forms so that the user can set/change its value or is the expectation that the applications will initialize this value with the current user?

**Comment by jcoyne** [comment]
I'm not sure adding depositor is required, see updates on #468

**Comment by hectorcorrea** [comment]
Updated with changes suggested on #468

**Comment by jcoyne** [comment]
:+1:

File: curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb [hunk 1]
```diff
@@ -7,7 +7,6 @@ module UploadSetBehavior
     included do
       has_many :file_sets, predicate: ActiveFedora::RDF::Fcrepo::RelsExt.isPartOf
 
-      property :creator, predicate: ::RDF::DC.creator
```
Comment (mjgiarlo) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: :+1: and I think we should replace with a marcrel-based depositor (and use the apply_depositor_metadata method on creation of UploadSets)

**Comment by mjgiarlo** [comment]
OK, left comments on #468.

File: curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb (hunk 1 above)
Comment (jcoyne) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: apply_depositor_metadata persists associated web acls into the repo. This will slow things down. Are you sure we want that?

Comment (mjgiarlo) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: What will slow things down? **Writing** those ACLs or their presence?

It seems broken to me that we have all these UploadSet objects just chilling out in Fedora without any association between them and their owners. Maybe this is an orthogonal concern that should be ticketed and discussed separately with more people?

Comment (jcoyne) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: There are associations between them and the files that are uploaded.  The files that are uploaded have the owners.  Can we infer their ownership through that?  UploadSets have no metadata, and are not editable, or deletable, they're basically just empty resources, used for a URL, right?  Why do we need to make them owned by anyone?

Comment (mjgiarlo) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: We can *kinda* infer ownership through the `belongs_to` relationship declared on FileSets, though proxy depositor/transfer functionality could introduce inconsistencies. If I deposit 10 files and then transfer n of them to someone else at some point, who gets to perform operations on the UploadSet? Me or someone else? Part of this is that I think having the UploadSets persist in the repository is a potential place to hang other functionality in the future. Perhaps that is YAGNI territory, and we could put a flag in this for figuring out at that point.

Comment (jcoyne) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: Sufia 6 is getting by just fine without, so I'd suggest not letting this PR get blocked.

Comment (mjgiarlo) [inline curation_concerns-models/app/models/concerns/curation_concerns/upload_set_behavior.rb]: I'm cool with that. I still have a question about wiping out the callback hooks above, which Sufia already uses to send notifications.

File: curation_concerns-models/app/jobs/upload_set_update_job.rb [hunk 2]
```diff
@@ -26,20 +26,8 @@ def perform(login, upload_set_id, title, file_attributes, visibility)
     upload_set.update(status: ["Complete"])
 
     if denied.empty?
-      unless saved.empty?
```
Comment (mjgiarlo) [inline curation_concerns-models/app/jobs/upload_set_update_job.rb]: I don't think we should wipe out the blocks of code in this method. Or, at very least, if we do, we should ticket adding them back. We use these callbacks in Sufia to send notifications.

Comment (jcoyne) [inline curation_concerns-models/app/jobs/upload_set_update_job.rb]: Ah, good catch, I didn't see that these are the descendants of `send_user_success_message` and `send_user_failure_message`

Comment (jcoyne) [inline curation_concerns-models/app/jobs/upload_set_update_job.rb]: I think we can restore the blocks and just use the `@login` rather than the depositor.

Comment (mjgiarlo) [inline curation_concerns-models/app/jobs/upload_set_update_job.rb]: :ok: #worksforme

**Comment by hectorcorrea** [comment]
I've restored the after hook events and updated them to use `@login` rather than the (now gone) `depositor`
